File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/sleep_timer/SleepTimerView.java
package com.github.anrimian.musicplayer.ui.sleep_timer;

import com.github.anrimian.musicplayer.domain.interactors.player.SleepTimerInteractor;

import moxy.MvpView;
import moxy.viewstate.strategy.AddToEndSingleStrategy;
import moxy.viewstate.strategy.StateStrategyType;

public interface SleepTimerView extends MvpView {

    @StateStrategyType(AddToEndSingleStrategy.class)
    void showSleepTimerTime(long sleepTimerTimeMillis);

    @StateStrategyType(AddToEndSingleStrategy.class)
    void showSleepTimerState(SleepTimerInteractor.SleepTimerState sleepTimerState);

    @StateStrategyType(AddToEndSingleStrategy.class)
    void showSleepRemainingSeconds(long remainingSeconds);
}

File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/sleep_timer/SleepTimerView.kt
package com.github.anrimian.musicplayer.ui.sleep_timer

import com.github.anrimian.musicplayer.domain.interactors.sleep_timer.SleepTimerState
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

interface SleepTimerView : MvpView {

    @AddToEndSingle
    fun showSleepTimerTime(sleepTimerTimeMillis: Long)

    @AddToEndSingle
    fun showSleepTimerState(state: SleepTimerState)

    @AddToEndSingle
    fun showRemainingTimeMillis(millis: Long)

}